File: B74560/Laboratorios/Lab2/Orbis/src/main/java/com/orbis/B74560/domain/Libro.java
package com.orbis.B74560.domain;

import java.util.Calendar;

public class Libro {

	private int id_libro;
	private String titulo;
	private short ano;
	private float precio;
	private int id_editorial;
	private final Calendar calendar = Calendar.getInstance();
	private final int anoActual = calendar.get(Calendar.YEAR);

	public Libro(int id_libro, String titulo, short ano, float precio, int id_editorial) {

		if (id_libro <= 0 || id_editorial <= 0 || ano <= 0 || precio <= 0)
			throw new RuntimeException("Hay datos imcompletos");
		if (ano > anoActual)
			throw new RuntimeException("Hay datos erroneos");
		if (titulo.trim().equals(""))
			throw new RuntimeException("Hay datos vacíos");

		this.id_libro = id_libro;
		this.titulo = titulo;
		this.ano = ano;
		this.precio = precio;
		this.id_editorial = id_editorial;

	}

	public Libro() {
	}

	public int getId_libro() {
		return id_libro;
	}

	public void setId_libro(int id_libro) {
		if (id_libro <= 0)
			throw new RuntimeException("Hay datos imcompletos");
		this.id_libro = id_libro;
	}

	public String getTitulo() {
		return titulo;
	}

	public void setTitulo(String titulo) {
		if (titulo.trim().equals(""))
			throw new RuntimeException("Hay datos vacíos");
		this.titulo = titulo;
	}

	public short getAno() {
		return ano;
	}

	public void setAno(short ano) {
		if ((ano > anoActual) && (ano <= 0))
			throw new RuntimeException("El año está erroneo");
		this.ano = ano;
	}

	public float getPrecio() {
		return precio;
	}

	public void setPrecio(float precio) {
		if (precio <= 0)
			throw new RuntimeException("Hay datos imcompletos");
		this.precio = precio;
	}

	public int getId_editorial() {
		return id_editorial;
	}

	public void setId_editorial(int id_editorial) {
		if (id_editorial <= 0)
			throw new RuntimeException("Hay datos imcompletos");
		this.id_editorial = id_editorial;
	}

}

File: B32915/OrbisB32915.Web/OrbisB32915.Web/Domain/Libro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace OrbisB32915.Web.Domain
{
    public class Libro
    {
        private int id_libro;
        private string titulo;
        private int ano;
        private double precio;
        private int isbn;
        private int id_editorial;

        public Libro(int id_libro, string titulo, int ano, double precio, int id_editorial, int isbn)
        {
        Id_libro = id_libro;
        Titulo = titulo;
        Ano = ano;
        Precio = precio;
        Isbn = isbn;
        Id_editorial = id_editorial;
    }
        public int Id_libro { get; internal set; }
        public string Titulo { get; internal set; }
        public int Ano { get; internal set; }
        public double Precio { get; internal set; }
        public int Isbn { get; internal set; }
        public int Id_editorial { get; internal set; }
    }
}
